hack/lib/test.sh

```diff
@@ -238,6 +238,24 @@ kube::test::if_has_string() {
   fi
 }
 
+kube::test::if_has_not_string() {
+  local message=$1
+  local match=$2
+
+  if [[ $(echo "$message" | grep "$match") ]]; then
```

Comment:
The original solution is bad for a large number of reasons. The command-line output of `grep` is not necessarily stable. if `grep` finds nothing on your system and in 2017, it doesn't output anything. Tomorrow, someone pushes a change that says "No matches found" (sound familiar? :) ) and suddenly the original check breaks.

The bash `if` statement is meant to check for return codes of commands (for example, in the original implementation it is checking for the return code of the `[[` command). `grep`, being a well-formed program, will always return `0` if and only if it succeeded, and will return non-zero otherwise. It is always better to use `if program` rather than spawning a subshell, capturing the output, relying on the hidden default behavior of `[[` with strings (that is different with `[`, by the way)...